src/main/java/siri/Parser.java

```diff
@@ -0,0 +1,226 @@
+package siri;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Class to make sense of user commands and execute the commands.
+ */
+class Parser {
+    private TaskList taskList;
+
+    /**
+     * Constructor of Parser class.
+     *
+     * @param taskList TaskList object in which execution of the commands will be carried out on.
+     */
+    Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Method to handle the command being passed in.
+     *
+     * @param command String representation of the command that is passed in.
+     * @return true to indicate further acceptance of command is allowed and false to indicate
+     * further acceptance of command is not allowed.
+     * @throws SiriException if command structure is invalid, command is not recognised or
+     * if there are missing or extra fields in command.
+     */
+    public String handleCommand(String command) throws SiriException {
```

Comment:
Perhaps extracting out logic for each case block would help with the long method?